sycl/unittests/Extensions/CommandGraph.cpp

```diff
@@ -2056,6 +2056,45 @@ TEST_F(CommandGraphTest, InvalidHostAccessor) {
   ASSERT_NO_THROW({ host_accessor HostAcc{Buffer}; });
 }
 
+TEST_F(CommandGraphTest, GraphPartitionsMerging) {
```

Comment:
Why do we only test backward dependencies?

Reply:
Backward dependencies involve potential merging of partitions which is not the case with forward-only dependencies.
e2e tests already test forward dependencies. This unitest was added to specifically test the merging part of the proposed algotithm.